File: app/src/main/java/org/antrack/app/libs/Net.java
package org.antrack.app.libs;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import java.io.IOException;

public class Net {
    public static boolean isOnline() {
        Runtime runtime = Runtime.getRuntime();
        try {

            Process ipProcess = runtime.exec("/system/bin/ping -c 1 8.8.8.8");
            int     exitValue = ipProcess.waitFor();
            return (exitValue == 0);

        } catch (IOException e)          { e.printStackTrace(); }
          catch (InterruptedException e) { e.printStackTrace(); }

        return false;
    }

    public static boolean isConnected(Context context) {
        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);

        if (cm == null) {
            return false;
        }

        NetworkInfo networkInfo = cm.getActiveNetworkInfo();

        if (networkInfo == null) {
            return false;
        }

        return networkInfo.isConnected();
    }
}

File: app/src/main/java/org/antrack/app/libs/Net.kt
package org.antrack.app.libs

import android.content.Context
import android.net.ConnectivityManager
import org.antrack.app.App
import java.io.IOException

object Net {
    val isOnline: Boolean
        get() {
            val runtime = Runtime.getRuntime()
            try {
                val ipProcess = runtime.exec("/system/bin/ping -c 1 8.8.8.8")
                val exitValue = ipProcess.waitFor()
                return exitValue == 0

            } catch (e: IOException) {
                e.printStackTrace()
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }

            return false
        }

    val isConnected: Boolean
        get() {
            val cm = App.context!!.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager ?: return false
            val networkInfo = cm.activeNetworkInfo
            return networkInfo.isConnected
        }
}
